dataRetriever/resolvers/validatorInfoResolver_test.go

```diff
@@ -361,6 +440,198 @@ func TestValidatorInfoResolver_ProcessReceivedMessage(t *testing.T) {
 		assert.Nil(t, err)
 		assert.True(t, wasCalled)
 	})
+
+	// resolveMultipleHashesRequest
+	t.Run("unmarshal fails", func(t *testing.T) {
+		t.Parallel()
+
+		numCalls := 0
+		expectedErr := errors.New("expected err")
+		args := createMockArgValidatorInfoResolver()
+		args.Marshaller = &testscommon.MarshalizerStub{
+			UnmarshalCalled: func(obj interface{}, buff []byte) error {
+				marshallerMock := testscommon.MarshalizerMock{}
+				if numCalls < 1 {
+					numCalls++
+					return marshallerMock.Unmarshal(obj, buff)
+				}
+
+				return expectedErr
+			},
+		}
+		res, _ := resolvers.NewValidatorInfoResolver(args)
+		require.False(t, check.IfNil(res))
+
+		err := res.ProcessReceivedMessage(createRequestMsg(dataRetriever.HashArrayType, []byte("hash")), fromConnectedPeer)
+		assert.Equal(t, expectedErr, err)
+	})
+	t.Run("no hash found", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgValidatorInfoResolver()
+		args.ValidatorInfoPool = &testscommon.CacherStub{
+			GetCalled: func(key []byte) (value interface{}, ok bool) {
+				return nil, false
+			},
+		}
+		args.ValidatorInfoStorage = &storage.StorerStub{
+			SearchFirstCalled: func(key []byte) ([]byte, error) {
+				return nil, errors.New("not found")
+			},
+		}
+		res, _ := resolvers.NewValidatorInfoResolver(args)
+		require.False(t, check.IfNil(res))
+
+		b := &batch.Batch{
+			Data: [][]byte{[]byte("hash")},
+		}
+		buff, _ := args.Marshaller.Marshal(b)
+		err := res.ProcessReceivedMessage(createRequestMsg(dataRetriever.HashArrayType, buff), fromConnectedPeer)
+		assert.True(t, strings.Contains(err.Error(), dataRetriever.ErrValidatorInfoNotFound.Error()))
```

Comment:
the IDE shows an issue `Method call 'err.Error()' may lead to nil pointer dereference` . add `assert.Error(t, err)` before this line